### PR TITLE
DB-11249 Remove unused ConstraintConstantAction

### PR DESCRIPTION
**Pull request by arnaud-splice**

### CODE REVIEW

**Review by dgomezferro — APPROVED** [review]

**Review by ascend1 — APPROVED** [review]

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/AlterTableNode.java [hunk 1]
```diff
@@ -524,22 +524,6 @@ public ConstantAction	makeConstantAction() throws StandardException
...
-                    // Check to see if a constraint of the same type
-                    // already exists
-                    ConstraintDescriptorList cdl =
-                        dd.getConstraintDescriptors(baseTable);
-
-                    if (cdl.getPrimaryKey() != null) {
-                        throw StandardException.newException(
-                            SQLState.LANG_ADD_PRIMARY_KEY_FAILED1,
-                            baseTable.getQualifiedName());
-                    }
```
Comment (martinrupp) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/AlterTableNode.java:540]: why has this been removed? looks like a check, is that check superflous?

**Review by martinrupp — CHANGES_REQUESTED** [review]

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/AlterTableNode.java (hunk 1 above)
Comment (arnaud-splice) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/AlterTableNode.java:540]: primaryKeyConstantActionCheck was checking if cca was of type ConstraintConstantAction, but that class was never used.

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/AlterTableNode.java (hunk 1 above)
Comment (martinrupp) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/AlterTableNode.java:540]: aaaahhh!

**Review by martinrupp — APPROVED** [review]

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by carolp-503 — APPROVED** [review]
+1

**Pull request merged**